client/src/components/MarkdownPage.js

```diff
@@ -16,7 +19,34 @@ export const MarkdownPage = ({
   markdown,
   width = 'large'
 }) => {
-  const config = {
+  const sectionId = getHash()
+  const [offset, setOffset] = useState(0)
+
+  useEffect(() => {
+    const sections = document.querySelectorAll(
+      'ol > li > p:first-child > span:first-child'
+    )
```

Comment:
Perhaps, this might be the easiest approach to handle this. Could you please tell me your approach of using `ref` in this case?

Reply:
We don't have to use `useRef` here, I recommended it as a solution. I would like to see this query selector scoped so that it will always query elements inside of this component and not the entire DOM.

Reply:
I've limited the scope to the `<main>` which renders the markdown page content at bf55be35ab7aa2af0c3bf618aab69e905e22828b.

Reply:
I've updated the `MarkdownPage` component to use the `useRef` hook for more explicit scoping at bc792d7b56685c254fe199794a3ac852680b07bf.